llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3414,9 +3415,14 @@ bool LoopVectorizationCostModel::interleavedAccessCanBeWidened(
   if (hasIrregularType(ScalarTy, DL))
     return false;
 
+  // We currently only know how to emit interleave/deinterleave with
+  // Factor=2 for scalable vectors.  This is purely an implementation
+  // limit.
+  if (VF.isScalable() && InterleaveFactor != 2)
```

Comment:
How is this handled for AArch64 SVE? Does the change mean support for different factors must be added for all architectures with scalable vectors at the same time?

Reply:
If there's the same check just later in LV, could it be replaced by an assert?

Reply:
> SVE also has a similar check in interleavedAccessCanBeWidened.
Indeed, it would be better to clean them up too if possible.

I can't find this check except in the AArch64 cost model (now removed).  What are you referring to?

Reply:
> Does the change mean support for different factors must be added for all architectures with scalable vectors at the same time?

This is an IR representation issue.  We have no way to represent in IR a interleave load or store at anything other than Factor=2 today.  When we change that, yes, we will have to audit the cost models for each of the targets which support the new intrinsic.  That seems pretty standard.

Reply:
> > SVE also has a similar check in interleavedAccessCanBeWidened.
> > Indeed, it would be better to clean them up too if possible.
> 
> I can't find this check except in the AArch64 cost model (now removed). What are you referring to?

My mistake. I mentioned the wrong function earlier; it should be getInterleavedMemoryOpCost rather than interleavedAccessCanBeWidened. 
And commit 3a5c127f98bd918114c473cf75a565e46022d032 has already fixed it, thank you.